src/entities/models/dto.py

```diff
@@ -86,6 +86,12 @@ def validate_fi(self) -> "FinancialInstitutionDto":
                 raise ValueError(
                     f"Invalid tax_id {self.tax_id}. FinancialInstitution tax_id must conform to XX-XXXXXXX pattern."
                 )
+        if self.lei:
+            match = re.match(r"^([a-zA-Z0-9]{20})", self.lei)
+            if not match:
+                raise ValueError(
+                    f"Invalid lei {self.lei}. FinancialInstitution lei can only contain letters and numbers."
```

Comment:
Include a mention of the 20 character limit in the error description here, in case they send 10 characters that are only letters and numbers and get confused ;)